### PR TITLE
made recipe good

### PR DESCRIPTION
**Pull request by Grayson34**

### CODE REVIEW

File: backend/recipe/GroceryList.py [hunk 1]
```diff
@@ -93,4 +93,4 @@ def purchased_item(id, item):
         #precondition: id as int and item as string
         #postcondition: removes item from grocery list, adds item to stock
         GroceryList.delete_item(id, item)
-        Stock.add_item(id, item)
\ No newline at end of file
+        Stock.Stock.add_item(id, item)
```
Comment (ryanmercado) [inline backend/recipe/GroceryList.py:96]: found we can fix this by importing "from recipe.Stock import Stock" - This implementation works too, just looks a tad buggard

File: student-hub-app/src/Recipe/GroceryList.js [hunk 2]
```diff
@@ -100,24 +124,30 @@ const GroceryList = () => {
...
+                    onChange={(e) => setNewItem({ name: e.target.value })}
+                />
+                <button className="add-item-button" onClick={addItem}>Add</button>
+            </div>
+            <ul className="grocery-items-list">
+                {groceryItems.map((item, index) => (
+                    <li key={index} className= "grocery-item" >
+                        <span>{item}</span>
+                        <button className="remove-item-button" onClick={() => removeItem(item)}>X</button>
+                        <button className="purchased-button" onClick={() => purchasedItem(item)}>purchased!</button>
```
Comment (ryanmercado) [inline student-hub-app/src/Recipe/GroceryList.js:145]: I couldn't get this button to move items to stock in my testing

File: student-hub-app/src/Recipe/RecipeList.js [hunk 3]
```diff
@@ -113,53 +115,67 @@ const RecipeList = () => {
...
+    const RecipePopup = ({ recipe, onClose }) => {
+        return (
+            <div className="recipe-popup">
+                <h3>{recipe.name}</h3>
+                <p><strong>Ingredients:</strong> {recipe.ingredients}</p>
+                <p><strong>Measurements:</strong> {recipe.measurements}</p>
+                <p><strong>Steps:</strong> {recipe.steps}</p>
+                <button onClick={onClose}>Close</button>
+            </div>
+        );
```
Comment (ryanmercado) [inline student-hub-app/src/Recipe/RecipeList.js:128]: Do we want a "add to grocery list" feature here?

**Review by ryanmercado — APPROVED** [review]
Looks good enough for demo. I put some thoughts about next steps, and bugs I ran into while testing. Lets merge though

**Review by cdewitt02 — APPROVED** [review]
Good for MVP demo

**Pull request merged**